### PR TITLE
Update to use new xcode version for framework Mac build_tests

### PR DESCRIPTION
**Pull request by keyonghan**
This is a follow up of https://github.com/flutter/infra/pull/320

Related issue: https://github.com/flutter/flutter/issues/72823

### CODE REVIEW

File: config/framework_config.star [hunk 1]
```diff
@@ -800,7 +800,7 @@ def framework_try_config():
                 "sdk_version": XCODE_VERSION,
             },
             "$flutter/osx_sdk": {
-                "sdk_version": XCODE_VERSION,
+                "sdk_version": NEW_XCODE_VERSION,
             },
         },
         caches = MAC_DEFAULT_CACHES,
```
Comment (godofredoc) [inline config/framework_config.star:806]: You also need to update the caches.

**Comment by jmagman** [comment]
Is this intended to fix all the build_test failures happening now, or should I file a new issue?
```
/opt/s/w/ir/cache/osx_sdk/XCode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.0.sdk/usr/lib/swift/Swift.swiftmodule/armv7-apple-ios.swiftinterface:21488:21: error: module 'Builtin' has no member named 'convertUnownedUnsafeToGuaranteed'
                   return try body(Builtin.convertUnownedUnsafeToGuaranteed(fakeBase,
                                   ^~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
               <unknown>:0: error: failed to load module 'Swift'
```
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8860163589006361024/+/steps/run_test.dart_for_build_tests_shard_and_subshard_0/0/stdout

**Comment by jmagman** [comment]
> Is this intended to fix all the build_test failures happening now, or should I file a new issue?

Actually mac_build_tests ran 7 times starting yesterday https://ci.chromium.org/ui/p/flutter/builders/prod/Mac%20build_tests/1173/overview and eventually succeeded, so maybe it was fixed by https://github.com/flutter/infra/pull/320

**Comment by godofredoc** [comment]
> s intended to fix all the build_test failures happening now, or should I file a new issue?

This is only happening on try now.

**Comment by godofredoc** [comment]
> > Is this intended to fix all the build_test failures happening now, or should I file a new issue?
> 
> Actually mac_build_tests ran 7 times starting yesterday https://ci.chromium.org/ui/p/flutter/builders/prod/Mac%20build_tests/1173/overview and eventually succeeded, so maybe it was fixed by #320

I fixed the problem yesterday night. The problem was that the ios 14.0 added to the xcode 11.6 was not compatible with the swift included on xcode 11.6.

File: config/framework_config.star (hunk 1 above)
Comment (keyonghan) [reply config/framework_config.star]: Done.

**Review by godofredoc — APPROVED** [review]

**Comment by jmagman** [comment]
> or should I file a new issue?

https://github.com/flutter/flutter/issues/72899

**Pull request merged**